CppCoreGuidelines.md

```diff
@@ -6215,49 +6214,44 @@ Unless there is an exceptionally strong reason not to, make `x = std::move(y); y
 
 ##### Reason
 
-If `x = x` changes the value of `x`, people will be surprised and bad errors can occur. However, people don't usually directly write a self-assignment that turn into a move, but it can occur. However, `std::swap` is implemented using move operations so if you accidentally do `swap(a, b)` where `a` and `b` refer to the same object, failing to handle self-move could be a serious and subtle error.
+People don't usually directly write a self-move assignment, but it can occur.  `std::swap` is implemented using
+move operations so if you accidentally do `swap(a, b)` where `a` and `b` refer to the same object, failing to 
+handle self-move could be a serious and subtle error.  At a minimum, a self-move should put the object into a
+valid but unspecified state which is the same guarantee provided by the standard library containers.  It is 
+not requred to leave the object unchanged.  
 
 ##### Example
 
-    class Foo {
-        string s;
-        int i;
+If all of the members of a type are safe for move-assignment, the default generated move-assignment 
+operator will be safe too.
+
+    X& operator=(X&& a) = default;
+
+Otherwise, the manually written move-assignment operater must be made safe for self-assignement.
+
+    class X {
     public:
-        Foo& operator=(Foo&& a);
+        X();
+        X& operator=(X&& a) noexcept;
         // ...
+        ~X() { delete owning_ptr; }
+    private:
+        T* owning_ptr;  // bad (See R.20) but used in the example because
```

Comment:
I couldn't think of a simple example that required a manual move-assignment that didn't go against something in the guidelines